alf/networks/projection_networks.py

```diff
@@ -136,15 +138,15 @@ def __init__(self,
                 ("When squashing the mean or scaling the distribution, bounds "
                  + "are required for the action spec!")
 
-            action_high = torch.as_tensor(action_spec.maximum)
-            action_low = torch.as_tensor(action_spec.minimum)
-            self._action_means = (action_high + action_low) / 2
-            self._action_magnitudes = (action_high - action_low) / 2
+            self._action_means, self._action_magnitudes = \
+                spec_utils.spec_means_and_magnitudes(action_spec)
             # Do not transform mean if scaling distribution
             if not scale_distribution:
-                self._mean_transform = (
-                    lambda inputs: self._action_means + self._action_magnitudes
-                    * inputs.tanh())
+                self._mean_transform = functools.partial(
```

Comment:
frankly speaking, I feel the previous version is better because of its readability.

Reply:
Changed was done upon request... I can convert it back

Reply:
Maybe still keep the lambda here. And the other changes are fine.

Reply:
Converted back for now as the it does not help much compared to the original code.